### PR TITLE
devicetree: remove wdt fixups

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Get rid of the fixups by managing the sample and test code using compatibles instead. Preserve compatibility with a 'watchdog0' alias.

If we ever need to add more code related to watchdogs, we can think about generalizing a zephyr,watchdog chosen node or increasing use of the watchdog0 alias in SoC dtsi files. But for now, it's not necessary.

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by erwango — APPROVED** [review]

File: samples/drivers/watchdog/src/main.c [hunk 1]
```diff
@@ -14,15 +14,37 @@
...
+#define WDT_NODE DT_INST(0, st_stm32_window_watchdog)
+#elif DT_HAS_COMPAT(st_stm32_watchdog)
+#define WDT_NODE DT_INST(0, st_stm32_watchdog)
+#elif DT_HAS_COMPAT(nordic_nrf_watchdog)
+#define WDT_NODE DT_INST(0, nordic_nrf_watchdog)
+#elif DT_HAS_COMPAT(espressif_esp32_watchdog)
+#define WDT_NODE DT_INST(0, espressif_esp32_watchdog)
+#elif DT_HAS_COMPAT(silabs_gecko_wdog)
+#define WDT_NODE DT_INST(0, silabs_gecko_wdog)
+#elif DT_HAS_COMPAT(nxp_kinetis_wdog32)
```
Comment (henrikbrixandersen) [inline samples/drivers/watchdog/src/main.c:34]: Was `nxp_kinetis_wdog` intentionally left out?

File: tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c [hunk 2]
```diff
@@ -62,14 +62,34 @@
...
+#elif DT_HAS_COMPAT(st_stm32_watchdog)
+#define WDT_NODE DT_INST(0, st_stm32_watchdog)
+#elif DT_HAS_COMPAT(nordic_nrf_watchdog)
+#define WDT_NODE DT_INST(0, nordic_nrf_watchdog)
+#elif DT_HAS_COMPAT(espressif_esp32_watchdog)
+#define WDT_NODE DT_INST(0, espressif_esp32_watchdog)
+#elif DT_HAS_COMPAT(silabs_gecko_wdog)
+#define WDT_NODE DT_INST(0, silabs_gecko_wdog)
+#elif DT_HAS_COMPAT(nxp_kinetis_wdog32)
+#define WDT_NODE DT_INST(0, nxp_kinetis_wdog32)
```
Comment (henrikbrixandersen) [inline tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c:83]: Same question here.

Comment (mbolivar-nordic) [reply tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c:83]: This shouldn't be changing any functionality. If you look at the fixups I've removed, wdog32 is the only one currently supported. Therefore I would leave adding additional compatibles not currently supported out of scope of this PR, unless I am missing something.

File: soc/arm/nxp_kinetis/ke1xf/dts_fixup.h [hunk 3]
```diff
@@ -16,7 +16,5 @@
 #define DT_ADC_1_NAME				DT_NXP_KINETIS_ADC12_40027000_LABEL
 #define DT_ADC_2_NAME				DT_NXP_KINETIS_ADC12_4003C000_LABEL
 
-#define DT_WDT_0_NAME				DT_NXP_KINETIS_WDOG32_40052000_LABEL
```
Comment (mbolivar-nordic) [inline soc/arm/nxp_kinetis/ke1xf/dts_fixup.h:19]: @henrikbrixandersen this is the only fixup for NXP I found in tree.

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**